### ISSUE TITLE
Slow Query Plans With 3.7.0 vs 3.4.0

### ISSUE THREAD

**Issue by jrsperry** [neo4j/graphql#2059]
**Describe the regression**
Query Planning for queries generated under 3.7.0 take much longer than queries generated under 3.4.0 (likely due to the use of subqueries in the return) on complex queries. 

**Last working version**

Worked up to version: 3.4.0 (tested on, as later versions had some other performance related bugs)

Stopped working in version: 3.7.0

**Type definitions**

```
type Person {
  name: String
  actedIn: [Movie!]! @relationship(type: "ACTED_IN", direction: OUT)
}

type Movie {
  title: String
  actors: [Person!]! @relationship(type: "ACTED_IN", direction: IN)
}
```

**To Reproduce**

1. Using Neo4j with the movies db loaded
2. Use Above Type Definitions
3. Execute the following complex graphql statement. 
```
{
  people(where:{
    name_STARTS_WITH: "Tom"
    actedIn_SOME: {
      title_STARTS_WITH: "Jerry"
      actors_SOME: {
        name_STARTS_WITH: "Bonnie"
        actedIn_SOME: {
          title_NOT: "Top Gun"
          actors_SOME: {
            name_CONTAINS: "Patricia"
            actedIn_NONE: {
              title: "The Matrix"
            }
          }
        }
      }
    }
  }){
    name
    actedIn(where: {title_STARTS_WITH: "Jerry"
      actors_SOME: {
        name_STARTS_WITH: "Bonnie"
        actedIn_SOME: {
          title_NOT: "Top Gun"
          actors_SOME: {
            name_CONTAINS: "Patricia"
            actedIn_NONE: {
              title: "The Matrix"
            }
          }
        }
      }}){
      title
      actors(where: {name_STARTS_WITH: "Bonnie"
        actedIn_SOME: {
          title_NOT: "Top Gun"
          actors_SOME: {
            name_CONTAINS: "Patricia"
            actedIn_NONE: {
              title: "The Matrix"
            }
          }
        }
      }){
        name
        actedIn(where: { title_NOT: "Top Gun"
          actors_SOME: {
            name_CONTAINS: "Patricia"
            actedIn_NONE: {
              title: "The Matrix"
            }
          } }){
          title
          actors(where:{ name_CONTAINS: "Patricia"
            actedIn_NONE: {
              title: "The Matrix"
            }}){
            name
          }
        }
      }
    }
  }
}
```

5. The query will likely take ~7 seconds to execute (the first time) due to long query planning. The cypher generated under the 3.4.0 library takes ~300ms to plan and execute. 

Regression Analysis/Description: 

The long query planning also appears to use more heap, although it's difficult to pinpoint how much more or why, when doing some testing I just noticed I was getting out of heap errors from my DB when running the 3.7.0 library in my application with the same config as the 3.4.0. I had to increase the neo4j heap considerably to get rid of the heap errors. 

The long query planning is particularly painful if the db determines it's time to replan/the query cache fills up. Generated queries appear to be plenty performant and in my case even had fewer db hits, the planning can be just brutal.

Also noteworthy is the size of the query plans on the 3.7.0 library is much larger than the 3.4.0 library. 

**System (please complete the following information):**

-   OS: macOs
-   Node.js version:  16.16.0

**Comment by darrellwarde** [comment]
@angrykoala, is this a duplicate of #2032?

**Comment by jrsperry** [comment]
After looking over that issue, it's likely that it could be the same issue, although I haven't run the queries to validate. I know that 3.6.0 for sure had some issues with performance that was fixed with 3.7.0 but that was related to a huge amount of db hits being generated.

**Comment by jrsperry** [comment]
I just redid that guys steps, there `where exists` problem is likely the biggest problem. I did see quite a bit longer query planning times for the return (when the return was large/complex as opposed to simple) in my examples, but those were also leveraging `where exists`. I believe this is a duplicate of 2032. i'll close the issue.
